src/core/components/AuthenticateButton/index.js

```diff
@@ -53,9 +55,17 @@ export const mapStateToProps = (state) => ({
 });
 
 export const mapDispatchToProps = (dispatch) => ({
-  handleLogOut() {
+  handleLogOut({ api }) {
+    const { location } = window;
+    if (config.get('apiHost') === `${location.protocol}//${location.host}`) {
```

Comment:
I'm probably going to branch off of this and start that. The API is still in review so there's no rush on merging this but it feels like it's probably better off as a separate patch to me.